### ISSUE TITLE
Ignore property modifications on params named "state"

### ISSUE THREAD

**Issue by openjck** [airbnb/javascript#2485]
The [`no-param-reassign` rule](https://github.com/airbnb/javascript/blob/4ee732639396362c6981b1aa4730934c3baa7919/packages/eslint-config-airbnb-base/rules/best-practices.js#L228-L246) allows property modifications for params with certain names, like `acc` and `response`. I would like to suggest that `state` is also allowed, for users of Redux Toolkit. The Redux Toolkit documentation [recommends this](https://redux-toolkit.js.org/usage/immer-reducers#linting-state-mutations).

**Comment by openjck** [comment]
Yep, it's only the `state.x =` pattern that I'm referring to. I've updated the description to be more clear.

**Comment by ljharb** [comment]
If there was a way to prevent reassignment but allow mutation, that'd be worth considering, but i believe the rule doesn't offer that level of granularity.
